pkg/component/controller/cplb_unix.go

```diff
@@ -290,16 +312,47 @@ func (k *Keepalived) generateKeepalivedTemplate() error {
 	return nil
 }
 
+func (k *Keepalived) watchReconcilerUpdates() {
+	// Wait for the supervisor to start keepalived before
+	// watching for endpoint changes
+	process := k.supervisor.GetProcess()
+	for process == nil {
+		k.log.Info("Waiting for keepalived to start")
+		time.Sleep(5 * time.Second)
+		process = k.supervisor.GetProcess()
+	}
```

Comment:
It might work without this loop, if the nil check would be moved into the update loop.

Reply:
> If you feel strongly about it, I will change it, but I prefer it this way.

Currently, the loop cannot be cancelled externally. That's why I figured that we could just inline the nil check below. Would be less code, too. I'm fine with keeping it, as long as it can be cancelled.

> it will just return a process with an invalid PID

Right. I'd argue that this is a current shortcoming of Supervisor, though.

Reply:
@twz123 I accidentally editted your comment with my reply 🤦 

The reply is: 

I added a limit of 6 times, that's 30 seconds and should be way more than enough time to start keepalived. It gets cancelled eventually, it's not the prettiest but should fix the problem.

Unfortunatelly I deleted your comment saying that it had to be possible to cancel it from the outside.